src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js

```diff
@@ -46,14 +47,16 @@ foam.CLASS({
 
   methods: [
     {
-      name: 'generateKey',
+      name: 'generateKeyAndQR',
       javaCode: `
+        Logger logger = (Logger) x.get("logger");
+
         User user = (User) (x.get("agent") != null ?
           x.get("agent") :
           x.get("user")) ;
         DAO userDAO = (DAO) x.get("localUserDAO");
 
-        // fetch from user dao to get secret key
+        // fetch from localUserDAO to get updated user before setting TwoFactorSecret
```

Comment:
Personally, I don't find these kinds of comments to be super helpful and would recommend removing comments that simply describe what the code is doing. Comments are important for explaining "why" code is doing what it's doing as oppose to what the code is doing. Describing what the code is doing means a developer has to remember to update these comments any time the code changes and that's often forgotten.

Just my two cents :)